### ISSUE TITLE
Division of notification formats: popup / profile vs email 

### ISSUE THREAD

**Issue by linc** [vanilla/vanilla#3111]
A client recently requested full-text email notifications (the comment/discussion body in the email). The way it's currently hacked in is to add a 'Story' to those activity items, which overrides the use of the headline in the email body.

The problem is that adding a 'Story' does the same to the popup & profile notifications, so a long comment is reproduced entirely, causing uselessly long notifications on the site.

Similarly, another client requested that the email notification subject lines be altered to simply include the category and discussion name. However, doing this obliterates all context even in the popups & on the profile - you lose key phrases like "X mentioned you in Y" in the email _body_, in the popup, and in their profile, just because they wanted the email subject to be different.

Short version: There's too much re-use of the same data in too many different contexts.
